### PR TITLE
Add github action for qe-tests

### PR DESCRIPTION
**Pull request by sebrandon1**
Adding QE runners into our CI.  Starting with the `lifecycle` suite of tests first.

### CODE REVIEW

File: .github/workflows/qe.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      SHELL: /bin/bash
+      DEBUG_TNF: true # More verbose output from the TNF
+      KUBECONFIG: '/home/tnf/.kube/config'
+      PFLT_DOCKERCONFIG: '/home/tnf/.docker/config'
+      TEST_TNF_IMAGE_NAME: quay.io/testnetworkfunction/cnf-certification-test
+      TEST_TNF_IMAGE_TAG: localtest
+      DOCKER_CONFIG_DIR: '/home/tnf/.docker'
+
+    steps:
+      # 1) Clone the QE repository
```
Comment (bnshr) [inline .github/workflows/qe.yaml:30]: You can delete these comments.

**Review by bnshr — APPROVED** [review]

File: .github/workflows/qe.yaml (hunk 1 above)
Comment (sebrandon1) [reply .github/workflows/qe.yaml:30]: Sounds good I will do that.

File: .github/workflows/qe.yaml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    strategy:
+      fail-fast: false
+      matrix: 
+        # suite: [accesscontrol, affiliatedcertification, lifecycle, networking, observability, platformalteration, performance, operator]
+        suite: [lifecycle]
+    env:
+      SHELL: /bin/bash
+      DEBUG_TNF: true # More verbose output from the TNF
+      KUBECONFIG: '/home/tnf/.kube/config'
+      PFLT_DOCKERCONFIG: '/home/tnf/.docker/config'
```
Comment (rdavid) [inline .github/workflows/qe.yaml:24]:
```suggestion
      PFLT_DOCKERCONFIG: /home/tnf/.docker/config
```

File: .github/workflows/qe.yaml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+      - name: Clone the QE repository
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ env.QE_REPO }}
+          path: cnfcert-tests-verification
+
+      # Setup is complete.  Time to run the QE tests.
+      - name: Run the tests
+        run: FEATURES=${{matrix.suite}} TNF_REPO_PATH=${GITHUB_WORKSPACE} TNF_IMAGE=${{env.TEST_TNF_IMAGE_NAME}} TNF_IMAGE_TAG=${{env.TEST_TNF_IMAGE_TAG}} make test-features
```
Comment (rdavid) [inline .github/workflows/qe.yaml:81]: KUBECONFIG and LOCAL_TESTING are missed.

Comment (sebrandon1) [reply .github/workflows/qe.yaml:81]: `KUBECONFIG` is already set in the env.

`LOCAL_TESTING` isn't needed as I'm setting `TNF_IMAGE` and `TNF_IMAGE_TAG` for envconfig to pick up.

**Comment by rdavid** [comment]
How does Github know it should be run in VM and not in Container? And why the action is in this repo and not in QE?

**Comment by sebrandon1** [comment]
Github knows where to schedule the work based on: `runs-on: qe-runner`.

And yup we'll add it to both repos eventually so changes going into the QE repo get instant feedback as well.

**Comment by rdavid** [comment]
And what is `qe-runner`? How do you define it?

**Review by rdavid — APPROVED** [review]

**Comment by sebrandon1** [comment]
The `qe-runner` label is a label I've given to our fleet of self-hosted runner VMs.  Any runner using that label is eligible to pick up one of those jobs and start running.

**Pull request merged**